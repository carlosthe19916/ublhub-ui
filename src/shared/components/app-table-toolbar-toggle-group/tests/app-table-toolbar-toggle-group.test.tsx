import React from "react";
import { shallow } from "enzyme";

import { AppTableToolbarToggleGroup } from "../app-table-toolbar-toggle-group";

describe("AppTableToolbarToggleGroup", () => {
  it("Renders without crashing", () => {
    const filtersValue = new Map<string, string[]>();

    filtersValue.set("name", ["carlos", "maria"]);
    filtersValue.set("description", ["engineer", "teacher"]);

    const wrapper = shallow(
      <AppTableToolbarToggleGroup
        options={[
          {
            key: "name",
            name: "Name",
          },
          {
            key: "description",
            name: "Description",
          },
        ]}
        filtersValue={filtersValue}
        onDeleteFilter={jest.fn()}
      >
        <span>children</span>
      </AppTableToolbarToggleGroup>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
