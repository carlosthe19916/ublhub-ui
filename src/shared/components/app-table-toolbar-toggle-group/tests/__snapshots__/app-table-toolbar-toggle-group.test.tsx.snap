// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTableToolbarToggleGroup Renders without crashing 1`] = `
<ForwardRef
  variant="filter-group"
>
  <ToolbarFilter
    categoryName={
      Object {
        "key": "name",
        "name": "Name",
      }
    }
    chips={
      Array [
        "carlos",
        "maria",
      ]
    }
    deleteChip={[Function]}
    deleteChipGroup={[Function]}
    key="0"
    showToolbarItem={false}
  />
  <ToolbarFilter
    categoryName={
      Object {
        "key": "description",
        "name": "Description",
      }
    }
    chips={
      Array [
        "engineer",
        "teacher",
      ]
    }
    deleteChip={[Function]}
    deleteChipGroup={[Function]}
    key="1"
    showToolbarItem={true}
  >
    <span>
      children
    </span>
  </ToolbarFilter>
</ForwardRef>
`;
